os/mm/mm_heap/mm_getheap.c

```diff
@@ -259,3 +259,48 @@ int mm_get_heapindex(void *mem)
 	/* The address does not lie in the user heaps */
 	return INVALID_HEAP_IDX;
 }
+
+static size_t mm_find_largest_freesize(struct mm_heap_s *heap)
+{
+	size_t largest_size = 0;
+	struct mm_freenode_s *fnode;
+	int i = 0;
+
+	for(i = 0; i < MM_NNODES; ++i) {
+		for (fnode = heap->mm_nodelist[i].flink; fnode && fnode->size; fnode = fnode->flink) {
+			if (largest_size < fnode->size) {
+				largest_size = fnode->size;
+			}
+		}
+	}
```

Comment:
This way looks full searching of free node. Is't it possible to check first meet node with descending order?

Reply:
Modified. Thanks.